### ISSUE TITLE
`MockReset` strategy is no longer honored for `@MockitoBean` and `@MockitoSpyBean`

### ISSUE THREAD

**Issue by quaff** [spring-projects/spring-framework#33941]
```java
@SpringJUnitConfig
class MockBeanTests {

	@MockitoBean
	// @MockBean works
	ContextRefreshedEventProcessor contextRefreshedEventProcessor;

	@Test
	void test() {
		then(contextRefreshedEventProcessor).should().process(any(ContextRefreshedEvent.class));
	}

	@Configuration
	static class Config {

		@Bean
		ContextRefreshedEventListener contextRefreshedEventListener(
				ContextRefreshedEventProcessor contextRefreshedEventProcessor) {
			return new ContextRefreshedEventListener(contextRefreshedEventProcessor);
		}

	}

	static class ContextRefreshedEventListener {

		private final ContextRefreshedEventProcessor contextRefreshedEventProcessor;

		public ContextRefreshedEventListener(ContextRefreshedEventProcessor contextRefreshedEventProcessor) {
			this.contextRefreshedEventProcessor = contextRefreshedEventProcessor;
		}

		@EventListener
		public void onApplicationEvent(ContextRefreshedEvent event) {
			contextRefreshedEventProcessor.process(event);
		}

	}

	interface ContextRefreshedEventProcessor {
		void process(ContextRefreshedEvent event);
	}
}

```

The test fails with `@MockitoBean` but works fine with Spring Boot's `@MockBean`.

I think it's caused by `@MockitoBean` resetting the mock before tests automatically.

**Comment by sbrannen** [comment]
Thanks for bringing this to our attention, @quaff! 👍 

I apologize: I inadvertently removed the `MockReset` check for that particular use case while refactoring `MockitoResetTestExecutionListener` in 6c2cba5d8ad29657a5164dd1ad369064f39660cf.

So, I'll add that back and make sure we have tests in place that cover this use case.

**Comment by quaff** [comment]
I confirm it's fixed now, thank you @sbrannen

**Comment by sbrannen** [comment]
> I confirm it's fixed now, 

Great! Thanks for letting us know, @quaff. 👍 

> thank you @sbrannen

You're welcome.

**Comment by jesperancinha** [comment]
@sbrannen , I'm afraid this isn't fixed yet. I made a pull request with your suggestion and it just doesn't build and it throws the same error: https://github.com/jesperancinha/jeorg-spring-test-drives/pull/737

**Comment by jesperancinha** [comment]
Thanks for your response @sbrannen ! You did only mention  `spring-context` and I just wanted to make sure that your next release would work. I see that it does and so I'm hopeful this won't be a problem on the next release. That's how much I care about the spring framework. Anyway, thank you for your input and for getting this issue fixed. I'll try to remember to remove the snapshot repo on the next release, but I'll probably forget it, and its not that important for one single module.

**Comment by sbrannen** [comment]
> Though I see that the [Spring Framework Artifacts](https://github.com/spring-projects/spring-framework/wiki/Spring-Framework-Artifacts) wiki page consistently uses `spring-context` as an example, and perhaps that is what mislead you.
> 
> So, I'll update that wiki page to provide greater clarity.

FYI: I addressed that in #34002.
